src/spikeinterface/extractors/cellexplorersortingextractor.py

```diff
@@ -1,136 +1,163 @@
+from __future__ import annotations
+
 import numpy as np
 from pathlib import Path
-from typing import Union, Optional
 
 from ..core import BaseSorting, BaseSortingSegment
 from ..core.core_tools import define_function_from_class
 
 
-try:
-    import scipy.io 
-    import hdf5storage
-    HAVE_SCIPY_AND_HDF5STORAGE = True
-except ImportError:
-    HAVE_SCIPY_AND_HDF5STORAGE = False
-
-
-PathType = Union[str, Path]
-OptionalPathType = Optional[PathType]  
-
-
 class CellExplorerSortingExtractor(BaseSorting):
     """
     Extracts spiking information from .mat files stored in the CellExplorer format.
-    Spike times are stored in units of seconds.
+    Spike times are stored in units of seconds so we transform them to units of samples.
+
+    The newer version of the format is described here:
+    https://cellexplorer.org/data-structure/
+
+    Whereas the old format is described here:
+    https://github.com/buzsakilab/buzcode/wiki/Data-Formatting-Standards
 
     Parameters
     ----------
-    spikes_matfile_path : PathType
-        Path to the sorting_id.spikes.cellinfo.mat file.
+    folder_path: str | Path
+        Path to the folder containing the .mat files from the session.
+    sampling_frequency: float | None, optional
+        The sampling frequency of the data. If None, it will be extracted from the files.
+    session_info_matfile_path: str | Path | None, optional
+        Path to the session info .mat file. If None, it will be inferred from the folder path.
     """
 
     extractor_name = "CellExplorerSortingExtractor"
-    installed = HAVE_SCIPY_AND_HDF5STORAGE
     is_writable = True
-    mode = "file"
-    installation_mesg = "To use the CellExplorerSortingExtractor install scipy and hdf5storage: \n\n pip install scipy  hdf5storage"
+    mode = "folder"
+    installation_mesg = "To use the CellExplorerSortingExtractor install scipy and h5py"
+
+    def __init__(
+        self,
+        folder_path: str | Path,
```

Comment:
Technically breaking back-compatability in switching from file to folder mode; is this actually necessary to accomodate different versions of the associated header file (sessionInfo.mat vs. sessions.mat)?

Reply:
If they keep changing the format it is easier to get the top folder and modify the internal function to find the necessary files. My currrent take is that the price of backwards compatibility is pay to built a less brittle approach.

Reply:
I'd actually say folder formats are way more brittle; they are very easy to break by simply moving the files around, which is something that has been observed previously

Whereas having different arguments for the different required (or optional) file paths enables more degrees of freedom for the user that aren't dependent on following a strict folder pattern

Reply:
You have a point there. I will think more about this.

Reply:
OK, I reverted to file mode. I still want to standarize this with the other sorters so I flaged the old name for the file `spikes_matfile_path` with a deprecation warning for six months. None of this should affect neuroconv though, as the API over there does expose `file_path` instead of the soon to be deprecated `spikes_matfile_path`.